test/unittest/blas2/blas2_gemv_test.cpp

```diff
@@ -145,6 +145,186 @@ TYPED_TEST(BLAS_Test, gemv_test_transposed) {
+
+  for (int i = 0; i < m; ++i) {
+    ASSERT_NEAR(c_v_gpu_result[i], c_v_cpu[i], prec);
+  }
+
+  ex.get_policy_handler().template deallocate<scalar_t>(m_a_gpu);
+  ex.get_policy_handler().template deallocate<scalar_t>(v_b_gpu);
+  ex.get_policy_handler().template deallocate<scalar_t>(v_c_gpu);
+}
+
+// Test for a non-unit alpha value
+REGISTER_PREC(float, 1e-4, gemv_test_legacy_alpha)
+REGISTER_PREC(double, 1e-8, gemv_test_legacy_alpha)
+
+TYPED_TEST(BLAS_Test, gemv_test_legacy_alpha) {
+  using scalar_t = typename TypeParam::scalar_t;
+  using ExecutorType = typename TypeParam::executor_t;
+  using TestClass = BLAS_Test<TypeParam>;
+  using test = class gemv_test_legacy_alpha;
+
+  int m = 125;
+  int n = 127;
+  int lda = m;
+  int incX = 1;
+  int incY = 1;
+  const char* t_str = "n";  // Testing the no transpose matrix
+  scalar_t prec = TestClass::template test_prec<test>();
+  scalar_t alpha = scalar_t(5.0);
+  scalar_t beta = scalar_t(1.0);
+
+  // Input matrix
+  std::vector<scalar_t> a_m(m * n);
+  // Input Vector
+  std::vector<scalar_t> b_v(n);
+  // output Vector
+  std::vector<scalar_t> c_v_gpu_result(m, scalar_t(0.0));
+  // output system vector
+  std::vector<scalar_t> c_v_cpu(m, scalar_t(0.0));
+  TestClass::set_rand(a_m, m * n);
+  TestClass::set_rand(b_v, n);
+
+  // SYSTEM GEMMV
+  reference_blas::gemv(t_str, m, n, alpha, a_m.data(), m, b_v.data(), incX,
+                       beta, c_v_cpu.data(), incY);
+
+  SYCL_DEVICE_SELECTOR d;
+  auto q = TestClass::make_queue(d);
+  Executor<ExecutorType> ex(q);
+  auto m_a_gpu = ex.get_policy_handler().template allocate<scalar_t>(m * n);
+  auto v_b_gpu = ex.get_policy_handler().template allocate<scalar_t>(n);
+  auto v_c_gpu = ex.get_policy_handler().template allocate<scalar_t>(m);
+  ex.get_policy_handler().copy_to_device(a_m.data(), m_a_gpu, m * n);
+  ex.get_policy_handler().copy_to_device(b_v.data(), v_b_gpu, n);
+  ex.get_policy_handler().copy_to_device(c_v_gpu_result.data(), v_c_gpu, m);
+  // SYCLGEMV
+  _gemv(ex, *t_str, m, n, alpha, m_a_gpu, m, v_b_gpu, incX, beta, v_c_gpu,
+        incY);
+  auto event =
+      ex.get_policy_handler().copy_to_host(v_c_gpu, c_v_gpu_result.data(), m);
+  ex.get_policy_handler().wait(event);
+
+  for (int i = 0; i < m; ++i) {
+    ASSERT_NEAR(c_v_gpu_result[i], c_v_cpu[i], prec);
+  }
+
+  ex.get_policy_handler().template deallocate<scalar_t>(m_a_gpu);
+  ex.get_policy_handler().template deallocate<scalar_t>(v_b_gpu);
+  ex.get_policy_handler().template deallocate<scalar_t>(v_c_gpu);
+}
+
+// Test that the lda value is working correctly
+REGISTER_PREC(float, 1e-4, gemv_test_legacy_lda)
+REGISTER_PREC(double, 1e-8, gemv_test_legacy_lda)
+
+TYPED_TEST(BLAS_Test, gemv_test_legacy_lda) {
+  using scalar_t = typename TypeParam::scalar_t;
+  using ExecutorType = typename TypeParam::executor_t;
+  using TestClass = BLAS_Test<TypeParam>;
+  using test = class gemv_test_legacy_lda;
+
+  int m = 125;
+  int n = 127;
+  int lda = m * 2;
```

Comment:
@RossBrunton  this lda is not valid in blas
 LDA must be at least
*           max( 1, m ).
*           Unchanged on exit.

Reply:
But m * 2 > max(1, m).

Reply:
max( 1, m ) this means that for m>1 lda cannot be bigger than m.

Reply:
The spec says "LDA must be at least max( 1, m ).", assuming m is 100, this would be "LDA must be at least 100".

LDA is the total number of rows of the matrix which A is a submatrix of, it doesn't make sense for the submatrix to be larger than the outer matrix.

Reply:
LDA is the leading dimension of Matrix A, if the matrix A is M*N and the matrix is colum major the LDA is M.